### PR TITLE
chore(query-bar): Update query bar flow types

### PR DESCRIPTION
**Pull request by ChenCodes**
Changes in this PR:
- Make columns a required property (defaults to empty Array)
- ConditionID type in ConditionType should be a string and not a number
- onInputChange callback shared between TextInput fields in ValueField.js helps check for the case where the user clears the field, should be sending an empty array back up instead of an array with an empty string in it
- Filter Button dropdown should close when the user presses on the Apply button and the conditions have all been filled in

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ChenCodes has signed the CLA. Thanks for the pull request!

File: src/features/query-bar/__tests__/FilterButton-test.js [hunk 1]
```diff
@@ -80,15 +80,15 @@ describe('feature/query-bar/components/filter/FilterButton', () => {
...
                     {
                         columnId: '1',
-                        id: 4,
+                        id: '4',
                         operator: LESS_THAN,
                         values: [],
                     },
                 ],
             },
         ].forEach(({ description, conditionId, conditions, value, expectedConditions }) => {
```
Comment (Visylvius) [inline src/features/query-bar/__tests__/FilterButton-test.js]: could you change this to a test.each?

File: src/features/query-bar/components/ColumnButtonOverlay.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ type State = {
 };
 
 type Props = {
-    columns?: Array<ColumnType>,
+    columns: Array<ColumnType>,
```
Comment (macrotim) [inline src/features/query-bar/components/ColumnButtonOverlay.js]: Optional.

File: src/features/query-bar/components/filter/FilterButton.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ type State = {
 };
 
 type Props = {
-    columns?: Array<ColumnType>,
+    columns: Array<ColumnType>,
```
Comment (macrotim) [inline src/features/query-bar/components/filter/FilterButton.js]: I believe this is optional. 

It is reasonable to render a QueryBar without a template selected which means columns is undefined.

File: src/features/query-bar/components/filter/ValueField.js [hunk 4]
```diff
@@ -30,7 +33,7 @@ const ValueField = ({ onChange, selectedValues, valueOptions, valueType }: Props
                         hideLabel
                         label="String input"
                         name="string field"
-                        onChange={e => onChange([e.target.value])}
+                        onChange={onInputChange}
```
Comment (macrotim) [inline src/features/query-bar/components/filter/ValueField.js]: Nice cleanup!

File: src/features/query-bar/__tests__/FilterButton-test.js [hunk 5]
```diff
@@ -11,166 +11,119 @@ describe('feature/query-bar/components/filter/FilterButton', () => {
...
+            columnId: '1',
+            id: '3',
+            operator: EQUALS,
+            values: [],
+        };
+        const expectedConditions = [{ columnId: '2', id: '3', operator: EQUALS, values: [] }];
+        test.each`
+            columnId | conditions
+            ${'2'}   | ${[condition]}
+        `('should set conditions with an object with column', ({ columnId, conditions }) => {
```
Comment (macrotim) [inline src/features/query-bar/__tests__/FilterButton-test.js]: Thanks for test.each!

What does this test description mean?

File: src/features/query-bar/__tests__/FilterButton-test.js [hunk 6]
```diff
@@ -11,166 +11,119 @@ describe('feature/query-bar/components/filter/FilterButton', () => {
...
-                });
-                wrapper.instance().handleOperatorChange(conditionId, value);
-
-                expect(wrapper.state('conditions')).toEqual(expectedConditions);
+        ];
+        const expectedConditions = [{ columnId: '1', id: '4', operator: LESS_THAN, values: [] }];
+        test.each`
+            conditionId | value
+            ${'4'}      | ${LESS_THAN}
+        `('should set conditions with an object with operator', ({ conditionId, value }) => {
```
Comment (macrotim) [inline src/features/query-bar/__tests__/FilterButton-test.js]: Similar comment.

File: src/features/query-bar/__tests__/FilterButton-test.js [hunk 7]
```diff
@@ -11,166 +11,119 @@ describe('feature/query-bar/components/filter/FilterButton', () => {
...
-                });
-                wrapper.instance().handleValueChange(conditionId, values);
-
-                expect(wrapper.state('conditions')).toEqual(expectedConditions);
+        ];
+        const expectedConditions = [{ columnId: '1', id: '5', operator: EQUALS, values: ['0'] }];
+        test.each`
+            conditionId | values
+            ${'5'}      | ${['0']}
+        `('should set conditions with an object with value', ({ conditionId, values }) => {
```
Comment (macrotim) [inline src/features/query-bar/__tests__/FilterButton-test.js]: Similar comment.

I assume the above unit test changes do not change functionality and only affect test.each and whitespace.

File: src/features/query-bar/__tests__/__snapshots__/FilterButton-test.js.snap [hunk 8]
```diff
@@ -21,7 +21,7 @@ exports[`feature/query-bar/components/filter/FilterButton render should render F
 >
   <Button
     className="query-bar-button"
-    isDisabled={true}
+    isDisabled={false}
```
Comment (macrotim) [inline src/features/query-bar/__tests__/__snapshots__/FilterButton-test.js.snap]: Expected?

Should the test description be rephrased to "should enable FilterButton when columns is non-empty"?

File: src/features/query-bar/__tests__/__snapshots__/FilterButton-test.js.snap [hunk 9]
```diff
@@ -66,7 +66,7 @@ exports[`feature/query-bar/components/filter/FilterButton render should render F
 >
   <Button
     className="query-bar-button"
-    isDisabled={false}
+    isDisabled={true}
```
Comment (macrotim) [inline src/features/query-bar/__tests__/__snapshots__/FilterButton-test.js.snap]: Same.

File: src/features/query-bar/__tests__/__snapshots__/FilterButton-test.js.snap [hunk 10]
```diff
@@ -45,7 +45,7 @@ exports[`feature/query-bar/components/filter/FilterButton render should render F
 </Flyout>
 `;
 
-exports[`feature/query-bar/components/filter/FilterButton render should render FilterButton when columns is not empty 1`] = `
+exports[`feature/query-bar/components/filter/FilterButton render should render FilterButton when columns is undefined 1`] = `
```
Comment (macrotim) [inline src/features/query-bar/__tests__/__snapshots__/FilterButton-test.js.snap]: Same.

File: src/features/query-bar/components/filter/FilterButton.js [hunk 11]
```diff
@@ -239,7 +239,7 @@ class FilterButton extends React.Component<Props, State> {
 
         const buttonClasses = classNames('query-bar-button', numberOfAppliedConditions !== 0 ? 'is-active' : '');
 
-        const isFilterDisabled = columns.length === 0;
+        const isFilterDisabled = columns === undefined || columns.length === 0;
```
Comment (macrotim) [inline src/features/query-bar/components/filter/FilterButton.js]: Should we cover columns === null?

If so, you can replace "columns === undefined" with "!columns".

File: src/features/query-bar/__tests__/FilterButton-test.js (hunk 7 above)
Comment (ChenCodes) [reply src/features/query-bar/__tests__/FilterButton-test.js]: Yep

File: src/features/query-bar/components/filter/FilterButton.js (hunk 11 above)
Comment (ChenCodes) [reply src/features/query-bar/components/filter/FilterButton.js]: Yes, it's possible that a developer could forget to pass in columns

**Comment by ChenCodes** [comment]
I'll do separate commits in future PRs.

**Pull request merged**